airflow/datasets/__init__.py

```diff
@@ -118,6 +118,23 @@ def coerce_to_uri(value: str | Dataset) -> str:
     return _sanitize_uri(str(value))
 
 
+def is_uri_normalized(uri: str) -> bool:
```

Comment:
Probably want to change the name to say `_is_uri_compliant` since we accept some non-normalised formats if they can be converted to a normalised format (trailing slashes etc.) so something that does not raise ValueError is not necessarily normalized. And you’d want to check the return value to get the actually normalized value.

Reply:
So, just to clarify, currently uri normalizers defined for some schemas (added in AIP-60) do not actually normalize the URI? Maybe we should change that by modifying the normalizers and adding some tests? 

The other option that you mentioned I can just change the name to `_is_uri_compliant` (why can't it be public?), and the when translating to OpenLineage we would have to deal with whatever is returned by the normalizer, without assuming the URI is normalized - so probably just adding more tests for different cases of the URI.

Reply:
This has been changed but I think what I was trying to say is, the way you implemented this function, it would return True for non-normalised URIs. The normalisation function always normalises the URI, but you didn’t use that normalised result, so the function only checks if the input _can_ be normalised, not if it _is already_ normalised as the name suggests.

Reply:
Yes, I agree, the combination of my naming and implementation was not correct.